Added ingestion in GTFS data to create route-station relationship

JPMartinezClaeys opened the pull request:
I added a few changes to ingest the table that maps each TransitStation object to 1+ TransitRoute objects:

- Included an ingestion restriction for uniqueness on the TransitStation + TransitRoute combination
- Added function in the file `extract_scheduled_gtfs.py` written by @mbjackson-capp for ingestion of the table

mbjackson-capp commented on app/scripts/extract_scheduled_gtfs.py:
```diff
@@ -63,7 +63,7 @@
 ALL_PILOT_CITY_URLS = [
     METRA_URL,
-    CTA_URL,
+    # CTA_URL,
```
Is there a reason CTA_URL got commented out?

mbjackson-capp commented:
Just to check, this creates a table in Postico where one column is filled with the _Postgres-assigned unique id number_ for a route (regardless of what route_id we ingested) and the other column is filled with the _Postgres-assigned unique id number_ for a station (regardless of what `station_id` we assigned; it's always an integer here, never a name). I take it the database uses those to relate the tables as quickly as possible
![Screen Shot 2024-05-13 at 11 42 28 AM](https://github.com/uchicago-capp-30320/RouteRangers/assets/111554739/ead98ba0-fa95-4331-91b8-0bb68fa37fc6)
![Screen Shot 2024-05-13 at 11 42 47 AM](https://github.com/uchicago-capp-30320/RouteRangers/assets/111554739/c01c0032-d7d5-4ad5-aa89-6d1e9a7c1535)



.
It looks like this didn't work for all systems/lines. For example, the station_id and route_id seem to be excluding the early id numbers corresponding to Metra.
Metra routes have Postgres-assigned id numbers from 1 to 11:
![Screen Shot 2024-05-13 at 11 37 45 AM](https://github.com/uchicago-capp-30320/RouteRangers/assets/111554739/fd6b3332-e446-4e41-8cd8-2c8a6b4dc414)
But the transitrouterelations start at 23, the lowest number for Portland:
![Screen Shot 2024-05-13 at 11 37 18 AM](https://github.com/uchicago-capp-30320/RouteRangers/assets/111554739/349dbe61-9fe9-425a-9742-6edec38c9fd3)
 
 
. 
Also, Metra stops have Postgres-assigned id numbers starting with 1:
![Screen Shot 2024-05-13 at 11 36 42 AM](https://github.com/uchicago-capp-30320/RouteRangers/assets/111554739/b26cfbff-b48e-4414-bc87-7e5468eb0bc2)

But the lowest station id number present in the transitrouterelation table is 965 (midway through New York):
![Screen Shot 2024-05-13 at 11 35 04 AM](https://github.com/uchicago-capp-30320/RouteRangers/assets/111554739/c5861d26-1a51-460d-8c2e-766b5c46752f)

Do you have a log of what happened when you ran the ingest for the first time, and either way do you know what (if any) exceptions occurred during ingestion?

mbjackson-capp commented:
Another thing to consider: Try using the `df.loc[:,"column"].str.strip()` method on each of the columns you `pd.merge` on. In  #62, @jimenasalinas and I discovered that there was a leading space for some columns in some GTFS files but not others, which caused merges to fail when you tried to merge, say, `BNSF_IB_1` and `<space>BNSF_IB_1`. Can we double check that something like that isn't causing issues here?

JPMartinezClaeys commented:
Thanks for the comment @mbjackson-capp I made the several fixes:
- Uncomment CTA url
- Added some cleaning before merging the `stops`, `trips` and `stop_times` Dataframes (this did fixed Metra ingestion)
- Added some error handling: Separated the error handling into (1) observations that have been already ingested and (2) other errors. For now, I'm just printing the error. (TODO: Add log with observations that could not be ingested)
